doc/migrations/v0.27-v.28.md

```diff
@@ -0,0 +1,343 @@
+
+Moving towards a separation of concerns regarding known peers' data, as well as enabling PeerStore persistence, the PeerStore is now divided into four main components: `AddressBook`, `ProtoBook`, `KeyBook` and `MetadataBook`. This resulted in API changes in the PeerStore, since each type of peer data should now be added in an atomic fashion. 
+
+### Add data
+
+**Before**
+```js
+const peerId = ...
+const peerInfo = new PeerInfo(peerId)
+
+peerInfo.protocols.add('/ping/1.0.0')
+peerInfo.protocols.add('/ping/2.0.0')
+peerInfo.multiaddrs.add('/ip4/127.0.0.1/tcp/0')
+
+libp2p.peerStore.put(peerInfo)
+```
+
+**After**
+```js
+const peerId = ...
+const protocols = ['/ping/1.0.0', 'ping/2.0.0']
+const multiaddrs = ['/ip4/127.0.0.1/tcp/0']
+
+libp2p.peerStore.protoBook.add(peerId, protocols)
+libp2p.peerStore.addressBook.add(peerId, multiaddrs)
+```
+
+### Get data
+
+**Before**
+```js
+const peerId = ...
+const peerInfo = libp2p.peerStore.get(peerId)
+// { id: PeerId, multiaddrs: MultiaddrSet, protocols: Set<string>}
+```
+
+**After**
+```js
+const peerId = ...
+const peer = libp2p.peerStore.get(peerId)
+// { id: PeerId, addresses: Array<{ multiaddr: Multiaddr }>, protocols: Array<string> }
+```
+
+### Has data
+
+**Before**
+```js
+const peerId = ...
+const hasData = libp2p.peerStore.has(peerId)
+```
+
+**After**
+```js
+const peerId = ...
+const hasData = Boolean(libp2p.peerStore.get(peerId))
+```
+
+### Delete data
+
+**Before**
+```js
+libp2p.peerStore.remove(peerId)
+```
+
+**After**
+```js
+// Atomic
+libp2p.peerStore.protoBook.delete(peerId)
+libp2p.peerStore.addressBook.delete(peerId)
+// Complete
+libp2p.peerStore.delete(peerId)
+```
+
+### Peers
+
+**Before**
+```js
+const peers = libp2p.peerStore.peers
+// Map<string, PeerInfo>
+```
+
+**After**
+```js
+const peers = libp2p.peerStore.peers
+// Similar to libp2p.peerStore.get()
+// Map<string, { id: PeerId, addresses: Array<{ multiaddr: Multiaddr }>, protocols: Array<string> }
+```
+
+## Migrating from Peer Info
+
+[`PeerInfo`][peer-info] is a libp2p peer abstraction layer that combines a [`PeerId`][peer-id] with known data of the peer, namely its multiaddrs and protocols. It has been used for a long time by `js-libp2p` and its modules to carry this data around the libp2p stack, as well as by the libp2p API, both for providing this data to the users or to receive it from them.
+
+Since this PeerInfo instances were navigating through the entire codebases, some data inconsistencies could be observed in libp2p. Different libp2p subsystems were running with different visions of the known peers data. For instance, a libp2p subsystem receives a copy of this instance with the peer multiaddrs and protocols, but if new data of the peer is obtained from other subsystem, it would not be updated on the former. Moreover, considering that several subsystems were modifying the peer data, libp2p had no visibility on which data was the updated and the older one.
```

Comment:
```suggestion
Since this PeerInfo instances were navigating through the entire codebases, some data inconsistencies could be observed in libp2p. Different libp2p subsystems were running with different visions of the known peers data. For instance, a libp2p subsystem receives a copy of this instance with the peer multiaddrs and protocols, but if new data of the peer is obtained from other subsystem, it would not be updated on the former. Moreover, considering that several subsystems were modifying the peer data, libp2p had no way to determine the accurate data.
```